src/Drawer/index.js

```diff
@@ -0,0 +1,232 @@
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+import React, { useRef, useEffect } from "react";
+import ReactDOM from "react-dom";
+import PropTypes from "prop-types";
+import cc from "classcat";
+import useMountTransition from "./useMountTransition";
+import useLockBodyScroll from "./useLockBodyScroll";
+
+const noop = () => {};
+
+/**
+ * Renders an animated drawer dialog with an overlay that
+ * allows navigation between multiple contents
+ *
+ * This component uses a `ReactDOM` portal to render it as a direct
+ * child of `document.body`.
+ */
+const Drawer = ({
+  isOpen = false,
+  onUserDismiss = noop,
+  onNext = null,
+  onPrev = null,
+  showControls = true,
+  children,
+  position = "right",
+  depth = "70%",
+  testId,
+}) => {
+  const shimRef = useRef(null);
+  const navRef = useRef(null);
+
+  const isTransitioning = useMountTransition(isOpen, 300);
+  const isHorizontal = position === "bottom" || position === "top";
+  // The depth is how far the drawer opens out, but the CSS prop depends
+  // on whether the Drawer is vertical or not
+  const depthStyle = isHorizontal ? { height: depth } : { width: depth };
+  useLockBodyScroll(isOpen);
+
+  // Navigation buttons on vertical drawers are the opposite compared to
+  // horizontal ones due to the order of navigation buttons being the opposite
+  //(rows are reversed in parent divs for horizontal drawers)
+  const prevOrNext = (opposite = false) => {
+    if (isHorizontal) {
+      if (opposite) return onPrev;
+      return onNext;
+    }
+    if (opposite) return onNext;
+    return onPrev;
+  };
+
+  const handleKeyDown = ({ key }) => {
+    if (key === "Escape") {
+      onUserDismiss();
+    }
+    if (isHorizontal) {
+      if (key === "ArrowRight") {
+        prevOrNext() && prevOrNext()();
+      }
+      if (key === "ArrowLeft") {
+        prevOrNext(true) && prevOrNext(true)();
+      }
+    } else {
+      if (key === "ArrowDown") {
+        prevOrNext(true) && prevOrNext(true)();
+      }
+      if (key === "ArrowUp") {
+        prevOrNext() && prevOrNext()();
+      }
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener("keydown", handleKeyDown);
+    return () => {
+      window.removeEventListener("keydown", handleKeyDown);
+    };
+  }, [handleKeyDown, isOpen]);
+
+  const handleShimClick = ({ target }) => {
+    if (
+      (navRef.current && target === navRef.current) ||
+      (shimRef.current && target === shimRef.current)
+    ) {
+      onUserDismiss();
+    }
+  };
+
+  /* eslint-disable jsx-a11y/no-static-element-interactions,jsx-a11y/click-events-have-key-events */
+  const navigationContainerJSX = (
+    <div
+      ref={navRef}
+      onClick={handleShimClick}
+      style={depthStyle}
+      className={cc([
+        `drawer navigation ${position}`,
```

Comment:
This is definitely the right approach to composing the classes here - for the sake of consistency, I'd like to change some of the class names to match our [naming conventions](https://github.com/narmi/design_system/blob/main/contributing/naming-conventions.md#css)

For example:
```diff
-.drawer.bottom
+.drawer--bottom
```

```diff
-.drawer.in
+.drawer--transitioning
```

This convention helps the component stay consistent with the naming in the rest of the project and also flattens the CSS specificity of each variant state to `10`